R/predict.R

```diff
@@ -51,10 +51,15 @@ predict.WrappedModel = function(object, task, newdata, subset = NULL, ...) {
     assertClass(task, classes = "Task")
     size = getTaskSize(task)
   } else {
-    assertDataFrame(newdata, min.rows = 1L)
+    ##FIXME: Better way to make into a data frame? This doesn't protect against a lot of things
     if (class(newdata)[1] != "data.frame") {
-      warningf("Provided data for prediction is not a pure data.frame but from class %s, hence it will be converted.",  class(newdata)[1])
-      newdata = as.data.frame(newdata)
+      if (xts::is.xts(newdata)) {
+        newdata = as.data.frame(newdata, row.names = as.character(index(newdata)))
+        assertDataFrame(newdata, min.rows = 1L)
```

Comment:
btw... as we mess here with `predict.WrappedModel`... would it be a good idea to have a `ForecastRegrModel`?